source/CIS/cis-4.rst

```diff
@@ -0,0 +1,788 @@
+
+
+Parameter
+~~~~~~~~~
+
+It is serialized as :ref:`CIS-4-SignatureEd25519` (``signature``) and message data ``RevocationDataOther`` consisting of :ref:`CIS-4-CredentialHolderId` (``credential_id``), metadata about the signature :ref:`CIS-4-SigningData` (``signing_data``), a revocation public key :ref:`CIS-4-PublicKeyEd25519` , and an optional revocation reason (``reason``), serialized similarly to :ref:`CIS-4-functions-revokeCredentialIssuer`::
+
+  RevocationDataOther ::= (credential_id: CredentialHolderId) (signing_data: SigningData) (revocation_key: PublicKeyEd25519) (reason: OptionalReason)
+  RevokeCredentialHolderParam ::= (signature: SignatureEd25519) (data : RevocationDataOther)
+
+
+Requirements
+~~~~~~~~~~~~
+
+- If revoked successfully, the credential status MUST change to ``Revoked`` (see :ref:`CIS-4-functions-credentialStatus`).
+- The message to be signed is produced in the following way:
+    - Start with the bytes of the domain separation string ``WEB3ID:REVOKE``.
+    - Append ``RevocationDataOther`` bytes from the input parameter.
+- The ``RevokeCredentialOtherParam``'s ``signing_data`` MUST include a nonce to protect against replay attacks.
+  The revocation authority's nonce is sequentially increased every time a revocation request is successfully executed.
+  The function MUST only accept a ``RevokeCredentialOtherParam`` if it has the next nonce following the sequential order.
+- The revocation MUST fail if any of the following conditions are met:
+    - The credential ID is not present in the registry.
+    - The revocation key in not present in the registry.
+    - The credential status is not one of ``Active`` or ``NotActivated`` (see :ref:`CIS-4-functions-credentialStatus`).
+    - The signature was intended for a different contract.
+    - The signature was intended for a different entrypoint.
+    - The signature is expired.
+    - The signature can not be validated.
+      The smart contract logic SHOULD practice its best efforts to ensure that only the revocation authority can generate and authorize a revocation request with a valid signature.
+
+.. _CIS-4-functions-registerRevocationKeys:
+
+``registerRevocationKeys``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Register public keys that can be used by revocation authorities.
+
+.. note::
+  This standard does not specify how the issuer is authenticated.
+  Implementations can use various mechanisms.
+  For example, the transaction sender's address is checked against the issuer's account address stored in the contract.
+  Another option is to use ``auxiliary_data`` to implement signature-based authentication mechanism.
+
+Parameter
+~~~~~~~~~
+
+It is serialized as First 2 bytes encode the length (``n``) of the vector of keys, followed by this many :ref:`CIS-4-PublicKeyEd25519` keys::
+
+  RegisterPublicKeyParameters ::= (n: Byte²) (key: PublicKeyEd25519)ⁿ
```

Comment:
In the `credential-registry` contract example, this also includes auxiliary data (I see you included it in the note, so maybe you just forgot to add it here as well?)
```suggestion
  AuxData ::= (n: Byte²) (data: Byteⁿ)
  RegisterPublicKeyParameters ::= (n: Byte²) (key: PublicKeyEd25519)ⁿ (auxiliary_data: AuxData)
```

Reply:
Yep, nice catch! Thanks!